molo/core/models.py

```diff
@@ -316,7 +316,8 @@ def save(self, *args, **kwargs):
 @receiver(
     post_save, sender=Image, dispatch_uid="create_image_info")
 def create_image_info(sender, instance, **kwargs):
-    if not hasattr(instance, 'image_info'):
+    if (not hasattr(instance, 'image_info') and
+            not ImageInfo.objects.get(image=instance)):
```

Comment:
This is the same (but with a hit to the db) as `if instance.image_info:`
I don't think you need to check the `hasattr`